### ISSUE TITLE
What is the reason behind binding listeners to global object?

### ISSUE THREAD

**Issue by anzome** [ai/nanoevents#20]
I'm pretty sure it was done intentionally. The comment there is also a hint of purposeful binding.

```js
i.apply(this, args) // this === global or window
```

However I still don't see the thing. On my level of understanding `i.apply(null, args)` is more predictable because it forces users to bind their listeners to the context they want on their own. Their listeners will throw an exception if they forget to do so. 

```js
window.test = 'test'
function testFn() { console.log(this.test); }
testFn.bind(window).apply(null); // outputs 'test', apply doesn't overwrite the context set with bind 
```
What I'm missing here?

https://github.com/ai/nanoevents/blob/master/index.js#L55

**Comment by ai** [comment]
Hm. Interesting idea. Will it reduce the library size?

Can you make PR and call `npx size-limit` to check size changes?

**Comment by anzome** [comment]
Ok. I'll prepare PR.

**Comment by anzome** [comment]
`i.apply(null, args)` by itself has no impact on the package size, it remains the 101 B. Of course, it doesn't nullify the main benefit of this change – provided callback relying of some context will fail for sure if user forget to bind it preliminary

But there is away to cut a single one byte out. The variable args declared there used only once down the road

```javascript
var args = [].slice.call(arguments, 1)
```

So it is possible to omit this medium variable and inline the slice call instead of the variable. Such solution looks reasonable since any reader of this code has to understand why the `[].slice.call()` construction used there.

```javascript
emit: function emit (event) {
  // Array.prototype.call() returns empty array if context is not array-like
  [].slice.call(this.events[event] || []).filter(function (i) {
    i.apply(null, [].slice.call(arguments, 1))
  })
}
```

$ npx size-limit

  Size limit:   101 B
  Package size: 100 B with all dependencies, minified and gzipped
  Loading time: 10 ms on slow 3G
  Running time: 29 ms on Snapdragon 410
  Total time:   39 ms

What do you think? Do you want such package size optimisation. Does it look motivated enough to you.

**Comment by anzome** [comment]
It was dump idea about size optimization. Please, forget about it. It will impact performance by multiple arguments slicing that will be done for each listener.

**Comment by ai** [comment]
`i.apply(null, [].slice.call(arguments, 1))` will slow down emitter :(

**Comment by ai** [comment]
Released in 2.0.0
